scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
Won't it make `safe_url_string` called twice? There are 2 issues with that: the first is performance (URL parsing can be a bottleneck); the second is that it'd stay in a way of https://github.com/scrapy/scrapy/issues/833